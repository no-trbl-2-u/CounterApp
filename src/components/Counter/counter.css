.count {
  font-size: 24px;
  background-color:black;
  color: white;
  margin: 9px;
}

.counter-container {
  margin: 20px;
  background-color:azure;
  width: 250px;
  margin:auto;
  border-radius: 15px;
  margin-top: 15px;
}

.buttons {
  margin: 10px;
}

.title {
  margin-left: 25px;
}