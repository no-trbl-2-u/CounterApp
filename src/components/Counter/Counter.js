import React, { Component } from 'react';
import './counter.css';

class Counter extends Component {
  constructor() {
    super();
    this.state = {
      value: 0
    };
    this.increment = this.increment.bind(this);
    this.decrement = this.decrement.bind(this);
    this.resetValue = this.resetValue.bind(this);
  }

  increment() {
    this.setState({
      value: this.state.value + 1,
    });
  }

  decrement() {
    this.setState({
      value: this.state.value === 0 ? 0 : this.state.value - 1,
    });
  }

  resetValue() {
    this.setState({
      value: 0,
    });
  }

  render() {
    return (
      <div className="counter-container">
        <span className="badge badge-success count">{this.state.value}</span>
        <button 
          className="btn btn-primary buttons"
          onClick={this.increment}> +
        </button>
        <button
          className="btn btn-danger buttons" 
          onClick={this.decrement}> -
        </button>
        <button
          className="btn btn-warning buttons" 
          onClick={this.resetValue}>Reset
        </button>
      </div>
    );
  }
}

export default Counter;
