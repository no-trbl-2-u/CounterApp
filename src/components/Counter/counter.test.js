import React from 'react';
import ReactDOM from 'react-dom';
import Counter from './Counter';
import renderer from 'react-test-renderer';

it('should render 0', () => {
  const tree = renderer.create(<Counter/>).toJSON();
  expect(tree).toMatchSnapshot();
})

it('should render to 2', () => {
  const component = renderer.create(<Counter/>);
  component.getInstance().increment();
  component.getInstance().increment();
  expect(component.toJSON()).toMatchSnapshot();
})

it('should not go negative', () => {
  const component = renderer.create(<Counter/>)
  
  // Add 1, Subtract 2, is it less than 0?
  component.getInstance().increment();
  
  component.getInstance().decrement();
  component.getInstance().decrement();

  expect(component.toJSON()).toMatchSnapshot();
})