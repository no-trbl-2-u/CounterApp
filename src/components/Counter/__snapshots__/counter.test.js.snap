// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not go negative 1`] = `
<div
  className="counter-container"
>
  <span
    className="badge badge-success count"
  >
    0
  </span>
  <button
    className="btn btn-primary buttons"
    onClick={[Function]}
  >
     +
  </button>
  <button
    className="btn btn-danger buttons"
    onClick={[Function]}
  >
     -
  </button>
  <button
    className="btn btn-warning buttons"
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;

exports[`should render 0 1`] = `
<div
  className="counter-container"
>
  <span
    className="badge badge-success count"
  >
    0
  </span>
  <button
    className="btn btn-primary buttons"
    onClick={[Function]}
  >
     +
  </button>
  <button
    className="btn btn-danger buttons"
    onClick={[Function]}
  >
     -
  </button>
  <button
    className="btn btn-warning buttons"
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;

exports[`should render to 2 1`] = `
<div
  className="counter-container"
>
  <span
    className="badge badge-success count"
  >
    2
  </span>
  <button
    className="btn btn-primary buttons"
    onClick={[Function]}
  >
     +
  </button>
  <button
    className="btn btn-danger buttons"
    onClick={[Function]}
  >
     -
  </button>
  <button
    className="btn btn-warning buttons"
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
